### PR TITLE
fix: don't coerce docker image tags from parameters to latest

### PR DESCRIPTION
**Pull request by loderunner**
https://circleci.atlassian.net/browse/DEVEX-1001

When the tag to a Docker image was just a parameter, the tag was silently coerced to `latest`. This led to an ambiguity in the diagnostic: "Docker image ... has no tag latest" (as the user did not specified `latest` anywhere).

This PR removes the coercion and leaves the identification of the parameter to the validation package.

E.g.

```yaml
  run_acceptance_tests:
    parameters:
      go_version:
        type: string
    docker:
      - image: cimg/go:<<parameters.go_version>>
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,9 +4,11 @@ go 1.19
 
 require (
 	github.com/Masterminds/semver v1.5.0
+	github.com/go-git/go-git/v5 v5.5.2
```
Comment (JulesFaucherre) [inline go.mod:7]: Hey no problem adding dependencies but I don't see any usage of these. Why did you add them?

File: pkg/parser/dockerImageParser.go [hunk 2]
```diff
@@ -39,11 +38,6 @@ func ParseDockerImageValue(value string) ast.DockerImageInfo {
 		tag = tag[1:]
 	}
 
-	// if the tag is a parameter, just consider that it's the "latest" tag for validation purposes
-	if utils.CheckIfOnlyParamUsed(tag) {
```
Comment (JulesFaucherre) [inline pkg/parser/dockerImageParser.go:43]: Nice to know it was just that 👍

**Review by JulesFaucherre — APPROVED** [review]
As of kind of E2E tests we have a file [`noErrors.yml`](https://github.com/CircleCI-Public/circleci-yaml-language-server/blob/37682a9289ad9b2e1f41999a40c46714bed1cf44/pkg/services/testdata/noErrors.yml) that is tested to have no errors and in which we put a lot of cases like this one, so you may want to add a line for your case

File: go.mod (hunk 1 above)
Comment (loderunner) [reply go.mod:7]: I didn't add anything. I think the tooling just reordered the dependencies by alphabetic order (see the removes lines below).

Comment (JulesFaucherre) [reply go.mod:7]: Ok my bad!
